tests/osn-tests/src/test_osn_global.ts

```diff
@@ -50,10 +57,92 @@ describe('osn_global', () => {
             try {
                 source = osn.Global.getOutputSource(1);
             } catch(e) {
-                throw new Error("failed to set source to output channel");
+                throw new Error(getCppErrorMsg(e));
             }
             
             expect(source.name).to.equal('test_osn_global_source');
         });
+
+        it('should fail if trying to get from empty output channel', () => {
+            let source: ISource;
+
+            try {
+                source = osn.Global.getOutputSource(5);
+            } catch(e) {
+                throw new Error(getCppErrorMsg(e));
+            }
+
+            expect(source).to.equal(undefined);
+        });
+    });
+
+    context('# GetOutputFlagsFromId', () => {
+        it('should get flags (capabilities) of a source type', () => {
+            let sourceType: string;
+            let sourceTypes: string[];
+            let flags: number = 0;
+
+            try {
+                sourceTypes = osn.InputFactory.types();
+            } catch(e) {
+                throw new Error(getCppErrorMsg(e));
+            }
+
+            expect(sourceTypes.length).to.not.equal(0);
+
+            for (sourceType of sourceTypes)
+            {
+                try {
+                    flags = osn.Global.getOutputFlagsFromId(sourceType);
+                } catch(e) {
+                    throw new Error(getCppErrorMsg(e));
+                }
+
+                expect(flags).to.not.equal(0);
+                flags = 0;
+            }
+        });
+    });
+
+    context('# LaggedFrames', () => {
+        it('should get lagged frames value', () => {
+            let laggedFrames: number;
+
+            expect(function() {
+                laggedFrames = osn.Global.laggedFrames;
+            }).to.not.throw();
```

Comment:
Good to test that but I'm not sure we are testing the correct behavior of the function here.

Reply:
Changed it. Now I initialize the value as undefined and after getting the value I check if it is still undefined.